### PR TITLE
feat(pnjs): enable commerce and add first course product

### PR DESCRIPTION
**Pull request by joelhooks**
courses

### CODE REVIEW

File: packages/commerce-next/src/pricing/pricing.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ const ProductImage = ({
 	return withImage ? (
 		<div
 			className={cn(
-				'bg-background dark:border-border dark:bg-background relative mx-auto -mb-32 h-56 w-56 rounded-full border border-gray-200 drop-shadow-xl',
```
Comment (joelhooks) [inline packages/commerce-next/src/pricing/pricing.tsx:132]: I assume this worked with proaws and needed the margin, but i was stumped and that project doesn't have an active product to compare with. I'm curious about customization with these components on a site to site basis, and what that might look like.

File: packages/core/src/schemas/product-schema.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const productSchema = z.object({
 		slug: z.string(),
 		image: z
 			.object({
-				url: z.string().url(),
+				url: z.string(),
```
Comment (joelhooks) [inline packages/core/src/schemas/product-schema.ts:20]: forms send this as an empty string which isn't a url. that should likely be filtered out, but at least this doesn't break.

File: apps/pro-nextjs/src/trpc/api/routers/contentResources.ts [hunk 3]
```diff
@@ -12,9 +12,9 @@ export const contentResourceRouter = createTRPCRouter({
 				.object({
 					contentTypes: z
 						.array(z.string())
-						.default(['event', 'lesson', 'tutorial']),
+						.default(['event', 'lesson', 'tutorial', 'workshop']),
 				})
-				.default({ contentTypes: ['event', 'lesson', 'tutorial'] }),
+				.default({ contentTypes: ['event', 'lesson', 'tutorial', 'workshop'] }),
```
Comment (joelhooks) [inline apps/pro-nextjs/src/trpc/api/routers/contentResources.ts:17]: new type, new filter

File: apps/pro-nextjs/src/lib/products-query.ts [hunk 4]
```diff
@@ -244,14 +244,17 @@ export async function updateProduct(input: Product) {
...
 	await db
 		.update(products)
 		.set({
 			name: input.name,
 			quantityAvailable: input.quantityAvailable,
 			status: 1,
 			fields: {
-				...input.fields,
+				...fieldsNoImage,
+				...(image?.url && { image }),
```
Comment (joelhooks) [inline apps/pro-nextjs/src/lib/products-query.ts:257]: addressing the empty string for the url

File: apps/pro-nextjs/src/app/(commerce)/products/[slug]/edit/_components/edit-product-form.tsx [hunk 5]
```diff
@@ -74,7 +74,9 @@ function ComboboxDemo({
 }) {
 	const [open, setOpen] = React.useState(false)
 
-	const { data = [] } = api.contentResources.getAll.useQuery()
+	const { data = [] } = api.contentResources.getAll.useQuery({
+		contentTypes: ['event', 'workshop'],
+	})
```
Comment (joelhooks) [inline apps/pro-nextjs/src/app/(commerce)/products/[slug]/edit/_components/edit-product-form.tsx:80]: filtering for the two types of paid products

File: apps/pro-nextjs/src/app/(commerce)/products/[slug]/edit/_components/edit-product-form.tsx [hunk 6]
```diff
@@ -163,18 +165,7 @@ export function EditProductForm({ product }: { product: Product }) {
...
-							}
-							return {
-								id: resourceItem.resource.id,
-								label: resourceItem.resource.fields?.title,
-								type: resourceItem.resource.type,
-								children: [],
-								itemData: resourceItem as any,
-							}
-						}),
+						children: [],
```
Comment (joelhooks) [inline apps/pro-nextjs/src/app/(commerce)/products/[slug]/edit/_components/edit-product-form.tsx:168]: this tree only needs to be a single level and we aren't trying to sort the product content at all here. doesn't need to be a tree at all, really.

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**